dynamiqs/time_array.py

```diff
@@ -156,12 +163,20 @@ def _factory_modulated(
             f' be a square matrix, but has shape {tuple(array.shape)}.'
         )
 
-    factors = [_ModulatedFactor(f, f0)]
+    factors = [_ModulatedFactor(Partial(f, args=args), f0)]
     arrays = array[None, ...]  # (1, n, n)
-    return ModulatedTimeArray(factors, arrays)
+    return ModulatedTimeArray(factors, arrays, static=jnp.zeros_like(array))
 
 
-class TimeArray:
+class SumCallable(eqx.Module):
```

Comment:
Looks like this could be a `CallableTimeArray`

Reply:
I don't think so. It's really just a trick to be able to sum callables while maintaining the jit. It was actually proposed by @patrick-kidger.